nominal/core/asset.py

```diff
@@ -2,18 +2,25 @@
 
 from dataclasses import dataclass, field
 from types import MappingProxyType
-from typing import Iterable, Mapping, Protocol, Sequence, cast
+from typing import Iterable, Literal, Mapping, Protocol, Sequence, cast
 
-from nominal_api import attachments_api, scout_asset_api, scout_assets, scout_run_api
-from typing_extensions import Self
+from nominal_api import (
+    scout_asset_api,
+    scout_assets,
+    scout_run_api,
+)
+from typing_extensions import Self, TypeAlias, deprecated
 
 from nominal.core._clientsbunch import HasAuthHeader
 from nominal.core._utils import HasRid, rid_from_instance_or_string, update_dataclass
 from nominal.core.attachment import Attachment, _iter_get_attachments
-from nominal.core.connection import Connection
+from nominal.core.connection import Connection, _get_connections
 from nominal.core.dataset import Dataset, _get_datasets
-from nominal.core.log import LogSet
-from nominal.core.video import Video
+from nominal.core.log import LogSet, _get_log_set
+from nominal.core.video import Video, _get_video
+
+ScopeTupleSeq: TypeAlias = "Sequence[tuple[str, Connection | Dataset | LogSet | Video]]"
+ScopeSeq: TypeAlias = "Sequence[Connection | Dataset | LogSet | Video | str]"
```

Comment:
what about just:

```suggestion
ScopeType: TypeAlias = Connection | Dataset | LogSet | Video
```

then using e.g. `Iterable[ScopeType | str]` or `Sequence[tuple[str, ScopeType]]` etc as needed?

Reply:
Yes, better!